lib/web_catalog/api_extractor.es6.js

```diff
@@ -0,0 +1,279 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const objectGraph = require('object-graph-js').ObjectGraph;
+
+var noise = false;     // True if base property like "caller", "length" will be returned.
+var showConst = false; // True if constant number/string/boolean value will be returned.
+
+var blacklistProperties = [
+  'toString',    // Do not visit object's toString.
+  'constructor',  // Do not visit constructor functions, they have prototype but not interface.
+  'CSS2Properties' // These are known bugs in Firefox.
+];
+
+var blacklistFunctionAttr = ['prototype'];
+// blacklistFunctionAttr is list of property of Function.prototype,
+// prototype is not a part of function, but we also wish to blacklist it.
+
+var blacklistObjectAttr = [];
+// blacklistObjectAttr is list of property of Object.prototype.
+
+var constType = ["boolean", "number", "string"];
+// Primitive types which is constant will be blacklisted.
+
+var rootId, ObjectId, FunctionId, objectId, functionId;
+
+// A wrapper function of og.getObjectKeys, constant will
+// be filtered if showConst is not turned on.
+function getObjectProperties(og, id) {
+  var keys = og.getObjectKeys(id);
```

Comment:
This function may return non-own keys. We have stored inherited properties in object graph, since they are not own properties of an object, they do not have metadata.